crates/bevy_input/src/gamepad/types.rs

```diff
@@ -0,0 +1,147 @@
+/// An array of every [`GamepadButtonType`].
+///
+/// ## Usage
+///
+/// This is used inside of the [`gamepad_event_system`](crate::gamepad::gamepad_event_system)
+/// to reset, set or remove the values of every [`GamepadButton`](crate::gamepad::GamepadButton).
+pub(crate) const ALL_BUTTON_TYPES: [GamepadButtonType; 19] = [
```

Comment:
IMO we should just derive the `EnumIter` trait from the `strum` crate instead. See #3419.